packages/website/src/components/magazine/index.js

```diff
@@ -1,28 +1,35 @@
-import React from 'react'
 import { getImage } from 'gatsby-plugin-image'
+import React from 'react'
+import { Heading } from '../../styles/sharedStyles'
 import ProjectCard from '../projects/components/ProjectCard/ProjectCard'
-import HeroSectionSlice from './hero-section-slice/HeroSectionSlice'
 import { CardWrapper, Wrapper } from './Elements'
-import { Heading } from '../../styles/sharedStyles'
+import HeroSectionSlice from './hero-section-slice/HeroSectionSlice'
 
 export default function MagazinePage ({ data }) {
   const date = data?.date
   const mainImage = getImage(data?.main_image)
   const title = data?.title?.text
   const info = data?.info?.text
   const link = data?.link?.url
+  const listTitle = data?.list_title?.text
+  const buttonName = data?.button_name?.text
+  const buttonTarget = data?.button_target ?? '_self'
+  const showButton = data?.show_button ?? false
```

Comment:
there are too many possible `undefined` variable definitions here, mainly arising because `data` can be `null/undefined`. 
Can we check for `data` inside caller for component (`MagzinePage`)?

Reply:
Makes sense, but will it make any difference 🤔? `Data` comes from `graphql query` & we can have any of those values `undefined` if those are empty in Prismic. Is there any possible fix you know of?

Reply:
It would make the code a lot cleaner.
I haven't worked with Prismic. So, I'm not sure how this is generally handled.

Reply:
In Prismic, if the document writer leaves any field empty it will be passed as `null` on querying it. Because of which we cannot predict which value will have `non-null` value.

Reply:
I think it should be better if we use some sort of runtime validations for this kind of stuff.
But lets consider it for some other PR.

Reply:
Sure thing 👍🏼